src/hotspot/share/oops/instanceKlass.hpp

```diff
@@ -31,6 +31,7 @@
 #include "oops/fieldInfo.hpp"
 #include "oops/instanceKlassFlags.hpp"
 #include "oops/instanceOop.hpp"
+#include "runtime/deoptimization.hpp"
```

Comment:
Can you just forward declare DeoptimizationScope rather than including the whole file here?

Reply:
Fixed, thanks! (not pushed yet)